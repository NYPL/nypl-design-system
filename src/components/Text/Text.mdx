import { ArgTypes, Canvas, Description, Meta } from "@storybook/blocks";

import * as TextStories from "./Text.stories";
import Link from "../Link/Link";

<Meta of={TextStories} />

# Text

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `0.25.1`   |
| Latest            | `1.7.1`    |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}

## Overview

<Description of={TextStories} />

The `Text` component renders a basic `<p>` element and can be used to render
text for differing content types, including multiple subtitle variants.

## Component Props

<Canvas of={TextStories.WithControls} />

<ArgTypes of={TextStories.WithControls} />

## Accessibility

The `Text` component renders paragraph `p` HTML elements which are inherently
accessible and will be read by screen readers.

Resources:

- [Chakra UI Text](https://chakra-ui.com/docs/components/typography/text)

## Updated Typographic Styles

New typographic styles were introduced in June 2023, adding `subtitle1`,
`subtitle2`, `overline1` and `overline2`. As part of that update, the `tag` and
`mini` styles have been deprecated.

For usage recommendations of each of the `size` options, please refer to the
[Typography Style
Guide](../?path=/docs/style-guide-typography--docs#text-display-sizes).

### Recommended Options

`body1`, `body2`, `caption`, `subtitle1`, `subtitle2`, `overline1`, `overline2`

<Canvas of={TextStories.RecommendedOptions} />

### Deprecated Options

`tag`, `mini`

<Canvas of={TextStories.DeprecatedOptions} />
