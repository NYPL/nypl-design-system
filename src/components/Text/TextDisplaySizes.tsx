export enum DisplaySizes {
  Default = "Default",
  Caption = "Caption",
  Small = "Tag",
  Mini = "Mini",
}
