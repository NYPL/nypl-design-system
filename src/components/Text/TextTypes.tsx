export enum TextTypes {
  Default = "Default",
  Bold = "Bold",
  Italic = "Italic",
  BoldItalic = "BoldItalic",
  Caption = "Caption",
  CaptionBold = "CaptionBold",
  CaptionItalic = "CaptionItalic",
  CaptionBoldItalic = "CaptionBoldItalic",
  Small = "Tag",
  SmallBold = "TagBold",
  SmallItalic = "TagItalic",
  SmallBoldItalic = "TagBoldItalic",
  Mini = "Mini",
  MiniBold = "MiniBold",
  MiniItalic = "MiniItalic",
  MiniBoldItalic = "MiniBoldItalic",
}
