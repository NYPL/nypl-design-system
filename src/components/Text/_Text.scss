.text {
  &--bold {
    font-weight: 700;
  }

  &--italic {
    font-style: italic;
  }

  &--bolditalic {
    font-style: italic;
    font-weight: 700;
  }

  &--caption {
    font-size: var(--font-size--1);
  }

  &--captionbold {
    font-size: var(--font-size--1);
    font-weight: 700;
  }

  &--captionitalic {
    font-size: var(--font-size--1);
    font-style: italic;
  }

  &--captionbolditalic {
    font-size: var(--font-size--1);
    font-style: italic;
    font-weight: 700;
  }

  &--small {
    font-size: var(--font-size--2);
    font-weight: 400;
  }

  &--smallbold {
    font-size: var(--font-size--2);
    font-weight: 700;
  }

  &--smallitalic {
    font-size: var(--font-size--2);
    font-style: italic;
    font-weight: 400;
  }

  &--smallbolditalic {
    font-size: var(--font-size--2);
    font-style: italic;
    font-weight: 700;
  }

  &--mini {
    font-size: var(--font-size--3);
    font-weight: 400;
  }

  &--minibold {
    font-size: var(--font-size--3);
    font-weight: 700;
  }

  &--miniitalic {
    font-size: var(--font-size--3);
    font-style: italic;
    font-weight: 400;
  }

  &--minibolditalic {
    font-size: var(--font-size--3);
    font-style: italic;
    font-weight: 700;
  }

  &--nospace {
    .nypl-ds & {
      margin: 0 0 0;
    }
  }
}
