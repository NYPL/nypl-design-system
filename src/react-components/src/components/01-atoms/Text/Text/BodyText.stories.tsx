import * as React from "react";

import BodyText from "./BodyText";
import { action } from "@storybook/addon-actions";

export default {
  title: "BodyText",
  component: BodyText,
};

export const text = () => <BodyText>Soufflé jujubes cupcake chocolate bar topping. Lollipop sesame snaps tootsie roll liquorice sweet chupa chups. Gummi bears tart sweet roll tiramisu biscuit cookie cake. Cookie chocolate jelly-o topping. Halvah bear claw wafer cupcake tiramisu ice cream tart gummi bears. Lemon drops chocolate cake croissant lemon drops gummies ice cream sugar plum. Sweet sesame snaps cookie toffee. Tart muffin bonbon jelly halvah jelly. Macaroon cake caramels jujubes macaroon lemon drops. Chocolate cake pie halvah. Macaroon lemon drops tiramisu chocolate bar jujubes tootsie roll apple pie. Icing sesame snaps pastry caramels muffin donut. Cake chocolate tiramisu marshmallow. Jujubes powder dragée caramels wafer apple pie bear claw apple pie.</BodyText>;
export const textWithLink = () => <BodyText>Soufflé jujubes chocolate bar topping. Lollipop sesame snaps tootsie roll liquorice sweet chupa chups. Gummi bears tart sweet roll tiramisu biscuit cookie cake. Cookie chocolate jelly-o topping. Halvah bear claw wafer cupcake tiramisu ice cream tart gummi bears. Lemon drops chocolate cake croissant lemon drops gummies ice cream sugar plum. Sweet sesame snaps cookie toffee. Tart muffin bonbon jelly halvah jelly. Macaroon cake caramels jujubes macaroon lemon drops. Chocolate cake pie halvah. Macaroon lemon drops tiramisu chocolate bar jujubes tootsie roll apple pie. Icing sesame snaps pastry caramels muffin donut. Cake chocolate tiramisu marshmallow. Jujubes powder dragée caramels wafer <a href="#apple-pie">apple pie</a> bear claw apple pie.</BodyText>;
export const textWithBold = () => <BodyText>Soufflé jujubes chocolate bar topping. Lollipop sesame snaps tootsie roll liquorice sweet chupa chups. Gummi bears tart sweet roll tiramisu biscuit cookie cake. Cookie chocolate jelly-o topping. Halvah bear claw wafer cupcake tiramisu ice cream tart gummi bears. Lemon drops chocolate cake croissant lemon drops gummies ice cream sugar plum. Sweet sesame snaps cookie toffee. Tart muffin bonbon jelly halvah jelly. Macaroon cake caramels jujubes macaroon lemon drops. Chocolate cake pie halvah. Macaroon lemon drops tiramisu chocolate bar jujubes tootsie roll apple pie. Icing sesame snaps pastry caramels muffin donut. Cake chocolate tiramisu marshmallow. Jujubes powder dragée caramels wafer <b>apple pie</b> bear claw apple pie.</BodyText>;
export const textWithItalic = () => <BodyText>Soufflé jujubes chocolate bar topping. Lollipop sesame snaps tootsie roll liquorice sweet chupa chups. Gummi bears tart sweet roll tiramisu biscuit cookie cake. Cookie chocolate jelly-o topping. Halvah bear claw wafer cupcake tiramisu ice cream tart gummi bears. Lemon drops chocolate cake croissant lemon drops gummies ice cream sugar plum. Sweet sesame snaps cookie toffee. Tart muffin bonbon jelly halvah jelly. Macaroon cake caramels jujubes macaroon lemon drops. Chocolate cake pie halvah. Macaroon lemon drops tiramisu chocolate bar jujubes tootsie roll apple pie. Icing sesame snaps pastry caramels muffin donut. Cake chocolate tiramisu marshmallow. Jujubes powder dragée caramels wafer <i>apple pie</i> bear claw apple pie.</BodyText>;
export const textWithOrderedList = () => <BodyText>Soufflé jujubes chocolate bar topping. <ol><li>Lollipop</li><li>sesame snaps</li> <li>tootsie roll</li></ol>  liquorice sweet chupa chups. Gummi bears tart sweet roll tiramisu biscuit cookie cake. Cookie chocolate jelly-o topping. Halvah bear claw wafer cupcake tiramisu ice cream tart gummi bears. Lemon drops chocolate cake croissant lemon drops gummies ice cream sugar plum. Sweet sesame snaps cookie toffee. Tart muffin bonbon jelly halvah jelly. Macaroon cake caramels jujubes macaroon lemon drops. Chocolate cake pie halvah. Macaroon lemon drops tiramisu chocolate bar jujubes tootsie roll apple pie. Icing sesame snaps pastry caramels muffin donut. Cake chocolate tiramisu marshmallow. Jujubes powder dragée caramels wafer apple pie bear claw apple pie.</BodyText>;
export const textWithUnorderedList = () => <BodyText>Soufflé jujubes chocolate bar topping. <ul><li>Lollipop</li><li>sesame snaps</li> <li>tootsie roll</li></ul>  liquorice sweet chupa chups. Gummi bears tart sweet roll tiramisu biscuit cookie cake. Cookie chocolate jelly-o topping. Halvah bear claw wafer cupcake tiramisu ice cream tart gummi bears. Lemon drops chocolate cake croissant lemon drops gummies ice cream sugar plum. Sweet sesame snaps cookie toffee. Tart muffin bonbon jelly halvah jelly. Macaroon cake caramels jujubes macaroon lemon drops. Chocolate cake pie halvah. Macaroon lemon drops tiramisu chocolate bar jujubes tootsie roll apple pie. Icing sesame snaps pastry caramels muffin donut. Cake chocolate tiramisu marshmallow. Jujubes powder dragée caramels wafer apple pie bear claw apple pie.</BodyText>;
export const textWithHeading = () => <BodyText><h2>Soufflé</h2><h3>jujubes</h3><h4>chocolate bar</h4><h5>topping</h5><h6>Lollipop</h6></BodyText>;
export const truncatedText = () => <BodyText maxchar={2}>Soufflé jujubes chocolate bar topping. Lollipop sesame snaps tootsie roll liquorice sweet chupa chups. Gummi bears tart sweet roll tiramisu biscuit cookie cake. Cookie chocolate jelly-o topping. Halvah bear claw wafer cupcake tiramisu ice cream tart gummi bears. Lemon drops chocolate cake croissant lemon drops gummies ice cream sugar plum. Sweet sesame snaps cookie toffee. Tart muffin bonbon jelly halvah jelly. Macaroon cake caramels jujubes macaroon lemon drops. Chocolate cake pie halvah. Macaroon lemon drops tiramisu chocolate bar jujubes tootsie roll apple pie. Icing sesame snaps pastry caramels muffin donut. Cake chocolate tiramisu marshmallow. Jujubes powder dragée caramels wafer apple pie bear claw apple pie.</BodyText>;
