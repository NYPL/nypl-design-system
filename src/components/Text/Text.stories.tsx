import type { Meta, StoryObj } from "@storybook/react";
import { withDesign } from "storybook-addon-designs";

import Text, { textSizesArray } from "./Text";
import Heading from "../Heading/Heading";
import Image from "../Image/Image";
import { Box } from "@chakra-ui/react";

const meta: Meta<typeof Text> = {
  title: "Components/Typography & Styles/Text",
  component: Text,
  decorators: [withDesign],
  argTypes: {
    className: { control: false },
    isBold: { table: { defaultValue: { summary: false } } },
    isItalic: { table: { defaultValue: { summary: false } } },
    isCapitalized: { table: { defaultValue: { summary: false } } },
    isUppercase: { table: { defaultValue: { summary: false } } },
    isLowercase: { table: { defaultValue: { summary: false } } },
    noSpace: { table: { defaultValue: { summary: false } } },
    size: {
      control: {
        type: "select",
      },
      options: textSizesArray,
      table: { defaultValue: { summary: "default" } },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Text>;

/**
 * Main Story for the Text component. This must contains the `args`
 * and `parameters` properties in this object.
 */
export const WithControls: Story = {
  args: {
    className: undefined,
    size: "default",
    isBold: false,
    isItalic: false,
    isCapitalized: false,
    isUppercase: false,
    isLowercase: false,
    noSpace: false,
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?type=design&node-id=10975-52&t=AEGDRpspvTyuSWSA-4",
    },
    jest: "Text.test.tsx",
  },
  render: (args) => (
    <>
      <Text {...args}>
        Animal Crossing[a] is a social simulation video game series developed
        and published by Nintendo. The series was conceptualized and created by
        Katsuya Eguchi and Hisashi Nogami. In Animal Crossing, the player
        character is a human who lives in a village inhabited by various
        anthropomorphic animals and can do various activities like fishing, bug
        catching, and fossil hunting. The series is notable for its open-ended
        gameplay and use of the video game console's internal clock and calendar
        to simulate real passage of time.
      </Text>
      <Text {...args}>
        Since its initial release in 2001, five Animal Crossing games have been
        released worldwide, one each for the Nintendo 64/iQue Player (enhanced
        and reissued for the GameCube), Nintendo DS, Wii, Nintendo 3DS and
        Nintendo Switch. The series has been both critically and commercially
        successful and has sold over 60 million units worldwide. Three spin-off
        games have also been released: Animal Crossing: Happy Home Designer for
        Nintendo 3DS, Animal Crossing: Amiibo Festival for Wii U, and Animal
        Crossing: Pocket Camp for mobile devices.
      </Text>
    </>
  ),
};

// The following are additional Heading example Stories.
export const SizeStyles: Story = {
  render: () => (
    <>
      <Text size="subtitle1">Subtitle 1</Text>
      <Text size="subtitle2">Subtitle 2</Text>
      <Text size="overline1">Overline 1</Text>
      <Text size="overline2">Overline 2</Text>
      <Text size="default">Default text</Text>
      <Text size="caption">Caption text</Text>
      <Text size="tag">Tag text</Text>
      <Text size="mini">Mini text (deprecated)</Text>
    </>
  ),
};
export const DefaultExample: Story = {
  render: () => (
    <>
      <Text>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent
        commodo cursus magna, vel scelerisque nisl consectetur et. Donec id elit
        non mi porta gravida at eget metus. Aenean lacinia bibendum nulla sed
        consectetur. Vestibulum id ligula porta felis euismod semper. Cras justo
        odio, dapibus ac facilisis in, egestas eget quam.
      </Text>
    </>
  ),
};
export const CaptionExample: Story = {
  render: () => (
    <>
      <Box width="fit-content">
        <Image src="//placekitten.com/400/400" size="medium" />
        <Text size="caption">This is a kitten</Text>
      </Box>
    </>
  ),
};
export const TagExample: Story = {
  render: () => (
    <>
      <Text size="tag">Tag text</Text>
    </>
  ),
};
export const MiniExample: Story = {
  render: () => (
    <>
      <Text size="mini">Mini text</Text>
    </>
  ),
};
export const Overline1Example: Story = {
  render: () => (
    <>
      <Text size="overline1" noSpace>
        Category
      </Text>
      <Heading size="heading2">Lorem ipsum dolor</Heading>
    </>
  ),
};
export const Overline2Example: Story = {
  render: () => (
    <>
      <Text size="overline2" noSpace>
        Category
      </Text>
      <Heading size="heading5">Lorem ipsum dolor</Heading>
    </>
  ),
};
export const Subtitle1Example: Story = {
  render: () => (
    <>
      <Heading size="heading2">Lorem ipsum dolor</Heading>
      <Text size="subtitle1" noSpace>
        Nullam id dolor id nibh ultricies vehicula ut id elit.
      </Text>
    </>
  ),
};
export const Subtitle2Example: Story = {
  render: () => (
    <>
      <Heading size="heading5">Lorem ipsum dolor</Heading>
      <Text size="subtitle2" noSpace>
        Nullam id dolor id nibh ultricies vehicula ut id elit.
      </Text>
    </>
  ),
};
