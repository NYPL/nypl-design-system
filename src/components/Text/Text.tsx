// HorizontalRule
import * as React from "react";
import bem from "../../utils/bem";
import { DisplaySizes } from "./TextDisplaySizes";

export interface TextProps {
  /** Optional additional attributes passed to the text element */
  attributes?: { [key: string]: any };
  /** Optional prop used to show bolded text */
  bold?: boolean;
  /** Optional className you can add in addition to `text` */
  className?: string;
  /** Optional prop used to show itlicized text */
  italic?: boolean;
  /** Optional modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
  /** Optional prop used to remove default spacing */
  noSpace?: boolean;
  /** Optional prop to control the text styling */
  displaySize?: DisplaySizes;
}

export default function Text(props: React.ComponentProps<"p"> & TextProps) {
  const {
    attributes = {},
    bold,
    children,
    className,
    displaySize,
    italic,
    modifiers = [],
    noSpace,
  } = props;

  if (displaySize && displaySize !== "Default") {
    modifiers.push(displaySize.toLowerCase());
  }

  if (bold) {
    modifiers.push("bold");
  }

  if (italic) {
    modifiers.push("italic");
  }

  if (noSpace) {
    modifiers.push("nospace");
  }

  return (
    <p className={bem("text", modifiers, "", [className])} {...attributes}>
      {children}
    </p>
  );
}
