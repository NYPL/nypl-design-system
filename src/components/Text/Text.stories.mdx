import {
  ArgsTable,
  Canvas,
  Description,
  Meta,
  Story,
} from "@storybook/addon-docs";
import { withDesign } from "storybook-addon-designs";

import Text from "./Text";
import { TextDisplaySizes } from "./TextTypes";
import { getCategory } from "../../utils/componentCategories";
import { getStorybookEnumValues } from "../../utils/utils";

export const enumValues = getStorybookEnumValues(
  TextDisplaySizes,
  "TextDisplaySizes"
);

<Meta
  title={getCategory("Text")}
  component={Text}
  decorators={[withDesign]}
  parameters={{
    design: {
      type: "figma",
      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/?node-id=10733%3A56919",
    },
  }}
  argTypes={{
    className: { control: false },
    displaySize: {
      control: { type: "select" },
      options: enumValues.options,
      table: { defaultValue: { summary: "TextDisplaySizes.Default" } },
    },
    isBold: { table: { defaultValue: { summary: false } } },
    isItalic: { table: { defaultValue: { summary: false } } },
    noSpace: { table: { defaultValue: { summary: false } } },
  }}
/>

# Text

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `0.25.1`   |
| Latest            | `0.26.0`   |

<Description of={Text} />

The `Text` component renders a basic `<p>` element. As its name indicates, the
`displaySize` prop controls the size of the text rendered in the component.

<Canvas withToolbar>
  <Story
    name="Text"
    args={{
      className: undefined,
      displaySize: "TextDisplaySizes.Default",
      isBold: false,
      isItalic: false,
      noSpace: false,
    }}
  >
    {(args) => (
      <>
        <Text {...args} displaySize={enumValues.getValue(args.displaySize)}>
          Animal Crossing[a] is a social simulation video game series developed
          and published by Nintendo. The series was conceptualized and created
          by Katsuya Eguchi and Hisashi Nogami. In Animal Crossing, the player
          character is a human who lives in a village inhabited by various
          anthropomorphic animals and can do various activities like fishing,
          bug catching, and fossil hunting. The series is notable for its
          open-ended gameplay and use of the video game console's internal clock
          and calendar to simulate real passage of time.
        </Text>
        <Text {...args} displaySize={enumValues.getValue(args.displaySize)}>
          Since its initial release in 2001, five Animal Crossing games have
          been released worldwide, one each for the Nintendo 64/iQue Player
          (enhanced and reissued for the GameCube), Nintendo DS, Wii, Nintendo
          3DS and Nintendo Switch. The series has been both critically and
          commercially successful and has sold over 60 million units worldwide.
          Three spin-off games have also been released: Animal Crossing: Happy
          Home Designer for Nintendo 3DS, Animal Crossing: Amiibo Festival for
          Wii U, and Animal Crossing: Pocket Camp for mobile devices.
        </Text>
      </>
    )}
  </Story>
</Canvas>

<ArgsTable story="Text" />
