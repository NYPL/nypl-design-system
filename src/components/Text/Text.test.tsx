import * as React from "react";
import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

import Text from "./Text";

describe("Text Accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(<Text>Animal Crossing is all that!</Text>);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Text", () => {
  it("renders a <p> tag and text", () => {
    render(<Text>Animal Crossing is all that!</Text>);
    expect(
      screen.getByText("Animal Crossing is all that!")
    ).toBeInTheDocument();
  });

  it("throws a warning when no children are passed", () => {
    const warn = jest.spyOn(console, "warn");
    render(<Text></Text>);
    expect(warn).toHaveBeenCalledWith(
      "NYPL Reservoir Text: No children were passed and the `Text` component " +
        "will not render correctly."
    );
  });

  it("renders the UI snapshot correctly", () => {
    const defaultText = renderer
      .create(<Text>Animal Crossing is all that!</Text>)
      .toJSON();
    const caption = renderer
      .create(<Text size="caption">Animal Crossing is all that!</Text>)
      .toJSON();
    const tag = renderer
      .create(<Text size="tag">Animal Crossing is all that!</Text>)
      .toJSON();
    const mini = renderer
      .create(<Text size="mini">Animal Crossing is all that!</Text>)
      .toJSON();
    const withChakraProps = renderer
      .create(
        <Text p="20px" color="ui.error.primary">
          Animal Crossing is all that!
        </Text>
      )
      .toJSON();
    const withOtherProps = renderer
      .create(<Text data-testid="props">Animal Crossing is all that!</Text>)
      .toJSON();

    expect(defaultText).toMatchSnapshot();
    expect(caption).toMatchSnapshot();
    expect(tag).toMatchSnapshot();
    expect(mini).toMatchSnapshot();
    expect(withChakraProps).toMatchSnapshot();
    expect(withOtherProps).toMatchSnapshot();
  });
});
