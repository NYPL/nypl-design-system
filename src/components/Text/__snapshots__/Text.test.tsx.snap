// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders the UI snapshot correctly 1`] = `
<p
  className="chakra-text css-0"
>
  Animal Crossing is all that!
</p>
`;

exports[`Text renders the UI snapshot correctly 2`] = `
<p
  className="chakra-text css-0"
>
  Animal Crossing is all that!
</p>
`;

exports[`Text renders the UI snapshot correctly 3`] = `
<p
  className="chakra-text css-0"
>
  Animal Crossing is all that!
</p>
`;

exports[`Text renders the UI snapshot correctly 4`] = `
<p
  className="chakra-text css-0"
>
  Animal Crossing is all that!
</p>
`;
